src/switchroot/ostree-prepare-root.c

```diff
@@ -214,8 +296,11 @@ main (int argc, char *argv[])
         composefs_mode = OSTREE_COMPOSEFS_MODE_MAYBE;
       else if (strcmp (ot_composefs, "on") == 0)
         composefs_mode = OSTREE_COMPOSEFS_MODE_ON;
-      else if (strcmp (ot_composefs, "signed") == 0)
-        composefs_mode = OSTREE_COMPOSEFS_MODE_SIGNED;
+      else if (strncmp (ot_composefs, "signed=", strlen ("signed=")) == 0)
+        {
+          composefs_mode = OSTREE_COMPOSEFS_MODE_SIGNED;
+          composefs_pubkey = ot_composefs + strlen ("signed=");
```

Comment:
Having the pubkey on the kernel commandline is probably fine, but longer term it feels like what we want instead is to have this baked into the initramfs as e.g. `/usr/lib/ostree/rootfs-initramfs-binding-key.pub` or so?

Basically in general I think kernel arguments should primarily be for things interpreted by the Linux kernel.  (Of course the historical use case of passing arguments to init (systemd) will probably always be there)

And this is actually argument that even `ostree=` should work this way.

Reply:
Do you mean, that instead of checking the commandline we just check for the existence of this file in the initrd, and if it is there we assume we need to validate against it?

Reply:
Right.

BTW one thing that might be nice here is to write the state into `/run/ostree-booted` - we could make that an `a{sv}` and write e.g. `composefs = "signed"` into it and that way e.g. `ostree admin status` and equivalent can show that the composefs was handled in a signed way (instead of these tools parsing kernel arguments, and also so they're explicitly verifying the post-execution state instead of the configuration-to-be-applied).